poc_iot_verifier/src/main.rs

```diff
@@ -57,27 +59,23 @@ impl Server {
         let pool = settings.database.connect(2).await?;
         sqlx::migrate!().run(&pool).await?;
 
-        // Create the density scaler query messaging channel
-        let (density_tx, density_rx) = density_scaler::query_channel(50);
-
         // configure shutdown trigger
         let (shutdown_trigger, shutdown) = triggered::trigger();
         tokio::spawn(async move {
             let _ = signal::ctrl_c().await;
             shutdown_trigger.trigger()
         });
 
+        let gateway_cache = GatewayCache::from_settings(settings).await?;
         let mut loader = loader::Loader::from_settings(settings).await?;
         let mut runner = runner::Runner::from_settings(settings).await?;
         let purger = purger::Purger::from_settings(settings).await?;
-        let mut density_scaler = DensityScaler::from_settings(settings.density_scaler.clone())?;
+        let mut density_scaler = DensityScaler::from_settings(settings.density_scaler.clone()).await?;
         tokio::try_join!(
-            runner.run(density_tx, &shutdown),
-            loader.run(&shutdown),
+            runner.run(&shutdown, &gateway_cache, density_scaler.hex_density_map()),
```

Comment:
ideally we could pass the hex density map by reference ?

Reply:
I we have to pass it in as is and clone it throughout because it is an [Arc](https://doc.rust-lang.org/std/sync/struct.Arc.html) and has to track where the pointer to the underlying map.